File: sdk/julia/src/billing.jl
# Billing & Contracts: balance, invoices, payment methods
struct BillingResource
    http::HttpClient
end

# GET /api/balance - Get account balance and credit
function get_balance(r::BillingResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/balance"; params = params)
end

# GET /api/invoice - List all invoices
function list_invoices(r::BillingResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/invoice"; params = params)
end

# GET /api/invoice/{id} - Get invoice details
function get_invoice(r::BillingResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/invoice/{id}"; params = params)
end

# POST /api/invoice/{id}/credit - Apply account credit to invoice
function apply_credit(r::BillingResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/invoice/{id}/credit"; params = params)
end

# GET /api/payment - List available payment methods
function get_payment_methods(r::BillingResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/payment"; params = params)
end

# GET /api/payment/fees - List payment methods with fees
function get_payment_fees(r::BillingResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/payment/fees"; params = params)
end

File: sdk/rust/src/billing.rs
//! Billing & Contracts: balance, invoices, payment methods

use std::sync::Arc;
use crate::http_client::HttpClient;

pub struct BillingResource {
    http: Arc<HttpClient>,
}

impl BillingResource {
    pub fn new(http: Arc<HttpClient>) -> Self { Self { http } }

    /// GET /api/balance - Get account balance and credit
    pub async fn get_balance(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/balance", None).await
    }

    /// GET /api/invoice - List all invoices
    pub async fn list_invoices(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/invoice", None).await
    }

    /// GET /api/invoice/{id} - Get invoice details
    pub async fn get_invoice(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/invoice/{id}", None).await
    }

    /// POST /api/invoice/{id}/credit - Apply account credit to invoice
    pub async fn apply_credit(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/invoice/{id}/credit", None).await
    }

    /// GET /api/payment - List available payment methods
    pub async fn get_payment_methods(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/payment", None).await
    }

    /// GET /api/payment/fees - List payment methods with fees
    pub async fn get_payment_fees(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/payment/fees", None).await
    }

}
